src/webui/api/torrentscontroller.cpp

```diff
@@ -909,14 +909,31 @@ void TorrentsController::addAction()
 void TorrentsController::addTrackersAction()
 {
     requireParams({u"hash"_s, u"urls"_s});
+    const QString hashParam = params()[u"hash"_s];
+    QList<BitTorrent::Torrent *> torrents;
+    const QList<BitTorrent::TrackerEntry> entries = BitTorrent::parseTrackerEntries(params()[u"urls"_s]);
 
-    const auto id = BitTorrent::TorrentID::fromString(params()[u"hash"_s]);
-    BitTorrent::Torrent *const torrent = BitTorrent::Session::instance()->getTorrent(id);
-    if (!torrent)
-        throw APIError(APIErrorType::NotFound);
+    const QStringList idStrings = hashParam.split(u'|', Qt::SkipEmptyParts);
+    for (const QString &hash : idStrings)
+    {
+        if (hash == u"*"_s)
+            continue;
+        const auto id = BitTorrent::TorrentID::fromString(hash);
+        BitTorrent::Torrent *const torrent = BitTorrent::Session::instance()->getTorrent(id);
+        if (!torrent)
+            throw APIError(APIErrorType::NotFound);
 
-    const QList<BitTorrent::TrackerEntry> entries = BitTorrent::parseTrackerEntries(params()[u"urls"_s]);
-    torrent->addTrackers(entries);
+        torrents.append(torrent);
+    }
+
+    if (hashParam == u"*"_s)
+    {
+        // add this tracker to all torrents
+        torrents = BitTorrent::Session::instance()->torrents();
+    }
+
+    for (BitTorrent::Torrent *const torrent : asConst(torrents))
+        torrent->addTrackers(entries);
```

Comment:
```suggestion
    const QList<BitTorrent::TrackerEntry> trackers = BitTorrent::parseTrackerEntries(params()[u"urls"_s]);
    const QStringList idStrings = params()[u"hashes"_s].split(u'|');
    applyToTorrents(idStrings, [&trackers](BitTorrent::Torrent *const torrent) { torrent->addTrackers(trackers); });
```

Reply:
Note that "all" is currently used as a placeholder for "all torrents".

Reply:
> Note that "all" is currently used as a placeholder for "all torrents".

I did not mean that "*" placeholder should be added too. I've just pointed out that it already exists, so there's no need for anything else.

Reply:
> Only thing I'm kinda wondering about is the removal of the 404 status ... do you think we need to keep that scenario ?

Personally, I've always considered it a bad idea to mix application logic with HTTP. HTTP layer should only be limited to transmitting requests/responses. Returning the HTTP status code in the cases like this one looks confusing (especially in the case of batch processing).
After all, the user just wants some action to be performed on some torrents. If some torrent does not exist (for example, it may be deleted between the time the torrent list was received and the batch request reached the server), then it is hardly a good idea to fail the entire batch (or even the rest of it).

@Chocobo1, what do you think about this change?

Reply:
>After all, the user just wants some action to be performed on some torrents. If some torrent does not exist (for example, it may be deleted between the time the torrent list was received and the batch request reached the server), then it is hardly a good idea to fail the entire batch (or even the rest of it).

I agree. Also, IMO ideally it should return a partial success code and probably list which failed (I'm not certain this is truly necessary).
Didn't this already discussed in the past?

>@Chocobo1, what do you think? Can we just remove support of * from removeTrackers action or support both all and * everiywhere?

I don't have strong opinion. Just make sure incompatible changes are announced properly.

Reply:
> Also, IMO ideally it should return a partial success code and probably list which failed (I'm not certain this is truly necessary).

Maybe within WebAPI v3 (when/if it is developed).

Reply:
> > @Chocobo1, what do you think? Can we just remove support of * from removeTrackers action or support both all and * everiywhere?
> 
> I don't have strong opinion. Just make sure incompatible changes are announced properly.

So maybe leave it as-is for now? I.e. accept `*` as a placeholder only where it is already accepted and `all` elsewhere. It can be easily done by replacing `*` with `all` before calling `applyToTorrents()`.

Reply:
>So maybe leave it as-is for now?

👍